### PR TITLE
[dune] Pack checker to avoid [odoc] problems + build doc for plugins.

### PR DESCRIPTION
**Pull request by ejgallego**
We also build documentation for plugins, for example `Ltac_plugin` is often used in other plugins.

### CODE REVIEW

File: checker/dune
Comment (Zimmi48) [inline checker/dune]: So we pack in Dune but not in make-based build, and thus we must open Checklib with a flag instead of modifying a source file. Is that correct? Why this choice?
When we start packing other parts like `kernel/` to have more control on the public API, we are going to have to pack it with both Dune and the make-based build (unless we wait to get rid of the make-based build first). And then, there will be an inconsistency.

**Review by Zimmi48 — APPROVED** [review]
My comment being a mere remark, I'll merge this PR if you confirm that you are positive about this approach.

File: checker/dune
Comment (ejgallego) [reply checker/dune]: That is correct. This choice is just an implementation of #8586 in Dune. As indeed packing is needed to workaround the `odoc` bug, it makes sense to do only so in Dune. Anyways would the make-based maintainers wanted to pack any library I am OK with it.

I am not sure why do you say there will be an inconsistency, can you explain more? [Checker is really a separate component]

I tried to pack indeed `kernel` using the make-based system, however I ran into problems so I prefer to wait for the make-based system to go away, as in particular using the dune version of packing will provide an easy migration path for plugins thanks to `(wrapped (transition ...))`.

**Pull request merged**
